Language/Java/Parser.hs

```diff
@@ -1107,8 +1100,10 @@ primaryNPS = try arrayCreation <|> primaryNoNewArrayNPS
 -- primaryNoNewArray = startSuff primaryNoNewArrayNPS primarySuffix
 
 primaryNoNewArrayNPS :: P (Exp Parsed, Location)
-primaryNoNewArrayNPS =
-  mapFst Lit <$> literal
+primaryNoNewArrayNPS = do
+  startLoc <- getLocation
+  endLoc <- getEndLoc
```

Comment:
Ich würde hier tatsächlich nicht `getEndLoc` verwenden, sondern `literal`, da diese Funktion schon die End-Location vom Literal mitliefert - sollte sich etwas an der Definition von `literal` etwas ändern in Zukunft (dass es aus mehr als einem Token besteht) , könnte es sein, dass hier nicht mehr die korrekte SourceSpan erfasst wird.

Reply:
Wir haben das nun so probiert
```
primaryNoNewArrayNPS :: P (Exp Parsed, Location)
primaryNoNewArrayNPS = do
  startLoc <- getLocation
  (_, endLoc) <- literal
  mapFst (Lit (startLoc, endLoc)) <$> literal
```
oder
```
primaryNoNewArrayNPS :: P (Exp Parsed, Location)
primaryNoNewArrayNPS = do
  startLoc <- getLocation
  (lit, endLoc) <- literal
  return (Lit (startLoc, endLoc) lit, endLoc)
```
Das führt aber dazu, dass der Parser nicht mehr funktioniert:
```
### Failure in: 0:MultipleStringLiterals:MultipleStringLiterals.java
jlint        > test/Tests.hs:26
jlint        > Parsing /Users/jokogr/src/jlint/test/java/MultipleStringLiterals.java failed with error:(line 2, column 20):
jlint        > unexpected IdentTok "a"
jlint        > Cases: 1  Tried: 1  Errors: 0  Failures: 1
jlint        > Test suite jlint-test failed
Completed 4 action(s).
```
bzw.
```
language-java>   transformer tests
language-java>     ExpressionName:               FAIL
language-java>       tests/TransformerTests.hs:32:
language-java>       (line 3, column 11):
language-java>       unexpected IdentTok "field"
```